pkg/operator/configobservation/controlplanereplicascount/observe_control_plane_replicas_count.go

```diff
@@ -0,0 +1,89 @@
+package controlplanereplicascount
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+var ControlPlaneReplicasCountPath = []string{"controlPlaneReplicasCount"}
+
+// ObserveControlPlaneReplicasCount sets the current control plane replicas count
+func ObserveControlPlaneReplicasCount(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains control-plane-replicas-count field.
+		ret = configobserver.Pruned(ret, ControlPlaneReplicasCountPath)
+	}()
+
+	// read the observed value
+	observedControlPlaneReplicasCountInt, err := readDesiredControlPlaneReplicasCount(genericListers.(configobservation.Listers).ConfigMapListerForKubeSystemNamespace)
+	if err != nil {
+		// we got an error, so we are unable to determine current replicas count
+		return existingConfig, append(errs, err)
+	}
+	// since unmarshalling JSON into an interface value always stores JSON number as a float64 convert the observed value
+	observedControlPlaneReplicasCount := float64(observedControlPlaneReplicasCountInt)
+
+	// read the current value
+	currentControlPlaneReplicasCount, _, err := unstructured.NestedFloat64(existingConfig, ControlPlaneReplicasCountPath...)
+	if err != nil {
```

Comment:
if this path doesn't exist, we don't get an error?  It's been a while, but I think you'll get an error on traversal errors and that isn't the same as "not a float".

Reply:
ah, you are right `NestedFloat64` returns `(float64, bool, error)`

Reply:
ah, sorry, this is actually correct, the current config is not required to have the key. This can be the very first time we need to set a value.